### PR TITLE
dependent mapping pattern (issue #1293)

### PR DESCRIPTION
**Pull request by guapi777**
dependent mapping pattern (issue #1293)

- Add dependent mapping pattern

Pull request description

- fix issue #1293

### CODE REVIEW

File: dependent-mapping/README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+---  
+layout: pattern  
+title: Dependent Mapping
+folder: dependent-mapping
+permalink: /patterns/dependent-mapping/  
+categories: Behavioral
+tags:
+- Database
```
Comment (iluwatar) [inline dependent-mapping/README.md:8]: The available tags are listed on page https://github.com/iluwatar/java-design-patterns/wiki/07.-Categories-and-Tags
I recommend using `Data access` here.

Comment (iluwatar) [reply dependent-mapping/README.md:8]: Please specify the language as well.

File: dependent-mapping/README.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+---  
+
+## Intent
+The basic idea behind Dependent Mapping is that one class (the dependent) relies upon some other class (the owner) for its database persistence. Each dependent can have only one owner and must have one owner.
+
+A dependent may itself be the owner of another dependent. In this case the owner of the first dependent is also responsible for the persistence of the second dependent. You can have a whole hierarchy of dependents controlled by a single primary owner.
+
+Writing and saving of dependents is left to the owner, and there are no outside references, updates to the dependents can be handled through deletion and insertion. Thus, if you want to update the collection of dependents you can safely delete all rows that link to the owner and then reinsert all the dependents.
```
Comment (iluwatar) [inline dependent-mapping/README.md:16]: Please provide the explanation section as well. See the other patterns how they should be formatted.

File: dependent-mapping/pom.xml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  <modelVersion>4.0.0</modelVersion>
+
+  <artifactId>dependent-mapping</artifactId>
+  <dependencies>
+    <dependency>
+      <groupId>org.testng</groupId>
+      <artifactId>testng</artifactId>
+      <version>RELEASE</version>
+      <scope>test</scope>
+    </dependency>
```
Comment (iluwatar) [inline dependent-mapping/pom.xml:19]: Is there a reason why we need TestNG and JUnit is not sufficient?

File: dependent-mapping/src/main/java/com/iluwatar/dependentmapping/Album.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+  /**
+   * Add track.
+   *
+   * @param track the track
+   */
+  void addTrack(Track track) {
+    tracks.add(track);
+  }
+}
```
Comment (iluwatar) [inline dependent-mapping/src/main/java/com/iluwatar/dependentmapping/Album.java:48]: Use Lombok annotations to reduce boilerplate

Comment (iluwatar) [reply dependent-mapping/src/main/java/com/iluwatar/dependentmapping/Album.java:48]: Check elsewhere too

File: dependent-mapping/src/main/java/com/iluwatar/dependentmapping/App.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    databaseAlbum1.addTrack(dataBaseTrack1);
+    databaseAlbum1.addTrack(dataBaseTrack2);
+    databaseAlbum2.addTrack(dataBaseTrack3);
+
+    Database database = new Database();
+    database.addAlbum(databaseAlbum1);
+    database.addAlbum(databaseAlbum2);
+    database.addTracks(dataBaseTrack1);
+    database.addTracks(dataBaseTrack2);
+    database.addTracks(dataBaseTrack3);
```
Comment (iluwatar) [inline dependent-mapping/src/main/java/com/iluwatar/dependentmapping/App.java:41]: Move the initialization to a separate method

File: dependent-mapping/src/main/java/com/iluwatar/dependentmapping/App.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+    AlbumMapper albumMapper = new AlbumMapper(database);
+    LOGGER.info("Test loading album");
+    Album album = albumMapper.loadAlbum(0);
```
Comment (iluwatar) [inline dependent-mapping/src/main/java/com/iluwatar/dependentmapping/App.java:45]: Use local variable type inference (var) where possible

File: dependent-mapping/src/main/java/com/iluwatar/dependentmapping/Database.java [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+   *
+   * @param trackName the track name
+   * @return the tracks
+   */
+  DataBaseTrack getTracks(String trackName) {
+    for (DataBaseTrack t : allTracks) {
+      if (t.getName().equals(trackName)) {
+        return t;
+      }
+    }
```
Comment (iluwatar) [inline dependent-mapping/src/main/java/com/iluwatar/dependentmapping/Database.java:42]: Use streams for more modern programming style

Comment (iluwatar) [reply dependent-mapping/src/main/java/com/iluwatar/dependentmapping/Database.java:42]: Check elsewhere too

**Review by iluwatar — CHANGES_REQUESTED** [review]

**Comment by santyelegy** [comment]
Thanks for the suggestions, we will fix them soon!

File: dependent-mapping/README.md (hunk 1 above)
Comment (guapi777) [reply dependent-mapping/README.md:8]: Excuse me, I want to know what "language" means? :)

**Pull request closed without merging**